internal/replication/blockreplicator.go

```diff
@@ -948,20 +952,43 @@ func (br *BlockReplicator) updateLastProposal(lastBlockProposed *types.Block) {
 			br.lg.Panicf("Failed to compute last block base hash: %s", err)
 		}
 		br.numInFlightBlocks++
+		if httputils.IsConfigBlock(lastBlockProposed) {
+			if br.inFlightConfigBlockNumber > 0 {
+				br.lg.Panicf("A config block was proposed, but there is still a config block in-flight! proposed: %d, in-flight: %d",
+					lastBlockProposed.GetHeader().GetBaseHeader().GetNumber(), br.inFlightConfigBlockNumber)
+			}
```

Comment:
Can't we, instead exiting on panic, somehow make it wait here, maybe for while?

Reply:
We could wait, but for what? the design makes sure that this never happens. If a config block was committed, then we should have been waiting on this exact number; not bigger or smaller number...